package.py

```diff
@@ -0,0 +1,911 @@
+        self.dist_directory = self.working_directory.joinpath('dist')
+        self.build_directory = self.working_directory.joinpath('build')
+        self.wheel_directory = self.build_directory.joinpath('wheels')
+        self.temporary_directory = self.build_directory.joinpath('temp')
+        self.backend_directory = self.build_directory.joinpath('backend')
+        self.build_directory.mkdir(parents=True, exist_ok=True)
+
+    def prepare_backend(self) -> None:
+        if self.backend_directory.exists():
+            shutil.rmtree(self.backend_directory)
+        self.backend_directory.mkdir(parents=True, exist_ok=True)
+
+    def prepare_temp(self) -> None:
+        self.temporary_directory.mkdir(exist_ok=True, parents=True)
+
+    def move_to_dist(self, file: Path) -> None:
+        self.dist_directory.mkdir(exist_ok=True)
+        logger.info(f'moving {file.name} to {self.dist_directory}')
+        shutil.move(src=file, dst=self.dist_directory)
+
+    def check_backend(self) -> None:
+        backend = self.backend_directory
+        if not backend.exists() or not any(backend.iterdir()):
+            logger.error(f'{backend} was missing or empty')
+            exit(1)
+
+    def copy_to_dist(self, file: Path) -> None:
+        self.dist_directory.mkdir(exist_ok=True)
+        logger.info(f'copying {file.name} to {self.dist_directory}')
+        shutil.copy(src=file, dst=self.dist_directory)
+
+
+class WindowsPackaging:
+    def __init__(self, storage: Storage, env: Environment) -> None:
+        self.__storage = storage
+        self.__env = env
+
+    @log_group('miniupnpc windows')
+    def setup_miniupnpc(self) -> None:
+        miniupnc = 'miniupnpc_64bit_py39-2.2.24.zip'
+        python_dir = Path(
+            subprocess.check_output(
+                'python -c "import os, sys; print(os.path.dirname(sys.executable))"',
+                encoding='utf8',
+                shell=True,
+            ).strip(),
+        )
+
+        if python_dir.name != 'Scripts':
+            python_dir = python_dir.joinpath('Scripts')
+
+        dll_filename = 'miniupnpc.dll'
+        dll_path = python_dir.joinpath(dll_filename)
+        if dll_path.exists():
+            logger.info(f'miniupnpc.dll is already installed in {python_dir}')
+            return
+
+        build_dir = self.__storage.build_directory
+        os.chdir(build_dir)
+        url = f'https://github.com/mrx23dot/miniupnp/releases/download/miniupnpd_2_2_24/{miniupnc}'
+        zip_filename = Downloader.download(url)
+        zip_path = build_dir.joinpath(zip_filename)
+        extraction_dir = build_dir.joinpath('miniupnpc')
+        extraction_dir.mkdir(exist_ok=True)
+        with ZipFile(zip_path, 'r') as zip_ref:
+            zip_ref.extractall(extraction_dir)
+
+        dll_file = extraction_dir.joinpath(dll_filename)
+        logger.info(f'moving {dll_file} to {python_dir}')
+
+        shutil.move(
+            src=dll_file,
+            dst=python_dir,
+        )
+
```

Comment:
```suggestion
```

Reply:
Is this suggestion about removing an empty line? shouldn't this already be taken care of by the linter?

Reply:
Yes. The linter doesn't do it and lefteris always makes me remove them